### PR TITLE
[MM-68749] Guard webContents finish-load handlers against destroyed objects

### PR DESCRIPTION
**Pull request by jgheithcock**
#### Summary

Several `webContents.once('did-finish-load' | 'did-frame-finish-load', ...)` handlers in the main process touched their underlying Electron object inside the callback body without checking `isDestroyed()` first. During `app.quit` (and other teardown races such as removing a popout/server view) a queued finish-load event can fire after the `WebContents`/`BrowserWindow` is destroyed; the handler then calls `send`, `focus`, `setTitle`, `show`, `getURL`, `loadURL`, sets `zoomLevel`, etc., and Electron throws `TypeError: Object has been destroyed`.

This shows up in Sentry shortly after a burst of `renderer.destroyed` events. Representative breadcrumb timeline from the linked Sentry issue:

| Time (Δms from quit) | Breadcrumb | Notes |
|---|---|---|
| -10000 | popout window + child renderers (loading screen, URL view, server view) created | user added a non-Mattermost server (404 on `/api/v4/system/ping`, `/api/v4/config/client`) |
| 0 | `window.closed` x2, `app.quit` | main window + popout closed |
| +2 → +33 | 6× `renderer.destroyed` | popout, loading screen, URL view, main window webContents all torn down |
| **+39** | **`TypeError: Object has been destroyed`** | a once-handler fires against a now-destroyed WebContents |

The non-Mattermost server is incidental: the partial-load state just makes pending finish-load events more likely. The same handlers can fire on any quit while a popout/loading screen is mid-load.

This PR applies the same `isDestroyed()` guard pattern already used elsewhere in the codebase (e.g. `webContentsManager.sendToAll`, `tray.update`, `MattermostWebContentsView.loadRetry`) to:

**Commit 1** — `did-finish-load` / `did-frame-finish-load` handler guards:

- `BaseWindow` — before setting `webContents.zoomLevel`
- `PopoutManager.startPopoutWindow` + `handleViewUpdated` — before `show` / `setTitle` / `send`
- `LoadingScreen.show` + `fade` — before `send` / `addChildView`
- `MattermostWebContentsView.useLastPath` — before `send(BROWSER_HISTORY_PUSH)` (also captures `lastPath` in a local since the field is cleared synchronously after `reload()`)
- `ModalView.show` — before `focus()`
- `MainWindow` — extends existing `!this.win` check with `isDestroyed()`
- `CallsWidgetWindow` — `did-frame-finish-load` `getURL()` was outside the existing `try`/`catch`

**Commit 2** — `PerformanceMonitor` defense-in-depth:

`PerformanceMonitor.registerView` / `registerServerView` use `webContents.on('did-finish-load', …)` to insert into the metrics maps. If this fires during teardown, the maps end up with a destroyed `WebContents` reference; the 60s `runMetrics` / `sendMetrics` interval would then call `send(...)` on it. Skip insertion at registration time and lazily drop destroyed entries when iterating.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-68749 (Sentry: https://mattermost-mr.sentry.io/issues/7393437427/)

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- ~[ ] Has UI changes~
- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
- [x] completed [Mattermost Contributor Agreement](https://mattermost.com/contribute/)
- [x] executed `npm run lint:js` for proper code formatting
- [x] Run E2E tests by adding label `E2E/Run`

#### Device Information
N/A — defensive guards in main-process code paths.

#### Release Note

```release-note
NONE
```

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/desktop/pull/3822)

## Change Impact: 🟠 Medium

**Regression Risk:** Changes add defensive guards across multiple window/view modules and the PerformanceMonitor. While they are narrowly-scoped (preventing accesses to destroyed Electron objects) and avoid altering public APIs or core business logic, the modifications span several modules (BaseWindow, PopoutManager, LoadingScreen, ModalView, MainWindow, CallsWidgetWindow, MattermostWebContentsView, PerformanceMonitor) and adjust lifecycle behaviors that can affect teardown and app-quit flows. Unit tests were added for key areas (PerformanceMonitor, MattermostWebContentsView, and several view/window tests) but some guarded paths remain simple and could lack exhaustive tests. Overall, moderate risk of regressions in teardown/finish-load sequencing or metrics reporting if edge cases are missed.

**QA Recommendation:** Perform targeted manual QA on app quit and popout/teardown flows, including:
- Quitting the app during page loads and verifying no "Object has been destroyed" crashes.
- Creating/closing popouts rapidly and ensuring no errors and correct titles/URLs.
- Validating PerformanceMonitor metrics behavior when views are destroyed mid-cycle.
Given existing automated tests cover many changes, a focused manual test pass is recommended rather than full regression.

*Generated by CodeRabbitAI*

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/desktop/pull/3822)

### CODE REVIEW

File: src/app/views/MattermostWebContentsView.ts [hunk 1]
```diff
@@ -291,8 +291,12 @@ export class MattermostWebContentsView extends EventEmitter {
             if (ViewManager.isPrimaryView(this.view.id)) {
                 this.webContentsView.webContents.send(BROWSER_HISTORY_PUSH, this.lastPath);
             } else {
+                const pathToPush = this.lastPath;
```
Comment (devinbinnie) [inline src/app/views/MattermostWebContentsView.ts:294]: Does this have to be pulled out? I don't think there's a case where this is changed between the call and the `did-finish-load` event.

**Review by devinbinnie — APPROVED** [review]
This bit about Electron is really annoying - first off, most of the `send()` calls should be no-ops if the web contents is destroyed, so I don't know why it has to throw an exception for these. Second, the fact that we have to destroy them in the first place and that they won't be automatically removed is why we have to deal with this at all.

Anyways, rant over. Just one non-blocking comment.

File: src/app/views/MattermostWebContentsView.ts (hunk 1 above)
Comment (jgheithcock) [reply src/app/views/MattermostWebContentsView.ts:294]: @devinbinnie - sorry, thought I had responded:

Yes, we need this. CodeRabbit picked this up and Opus agreed (I envision a Penguin whenever I say that). Line 292 doesn't have this issue as it is synchronous but the `once('did-finish-load')` is just registered, line 303 sets `lastPath` to `undefined` and some ms later, the `once` handler runs.
